File: java/CRenObj.java
package cell;
class DfCRenObj
{
	public static int CObject = 0;
	public static int CMolecule = 1;
	public static int Count = 2;
	public static int Null = 3;
};
public class CRenObj implements ISerialize
{
	int m_offset;
	protected String m_key="";
	protected String m_keyChange="";
	protected boolean m_remove=false;
	protected boolean m_show = true;
	public boolean GetRemove() { return m_remove;	 }
	public String GetKeyChange() {	return m_keyChange; }
	public void ClearKeyChange() { m_keyChange = new String();	 }
	public void SetShow(boolean _show) {	m_show=_show;	}
	public boolean GetShow()	{	return m_show;	}
	
	
	public String GetKey() { return m_key; }
	public void SetKey(String _key)
	{
		if (m_key.equals(_key))
			return;
		if(m_key.isEmpty()==false)
			m_keyChange = m_key;
		m_key = _key;
	}
	static int ls_off=0;
	public CRenObj()
	{
		
		m_offset = ls_off;
		ls_off++;
	}

	
	public void Render(CCamera _cam) {};
	public void Update(int _delay) {};
	public void LastUpdate(int _delay) {};
	public int GetCRenObjType() {return DfCRenObj.Null;	};
	
	public void Destroy() {};
	public void CanvasDelete() {};
	public int GetOffset() { return m_offset;	 }

	public CPacket Serialize()
	{
		CPacket pac=new CPacket();
		pac.Push(m_offset);
		pac.Push(m_key);

		return pac;
	}
	public void Deserialize(String _str)
	{
		CPacket pac=new CPacket();
		pac.Deserialize(_str);
		m_offset = pac.GetInt32(0);
		m_key=pac.GetString(1);
	}
};

File: js/CRenObj.js
function JSDfCRenObj()
{
	this.CObject = 0;
	this.CMolecule = 1;
	this.Count = 2;
	this.Null = 3;
};
var DfCRenObj=new JSDfCRenObj();

var g_RenObjOff=0;
class CRenObj extends ISerialize 
{
	constructor()
	{
		super();
		this.m_key="";
		this.m_keyChange="";
		this.m_offset=g_RenObjOff++;
		this.m_show=true;
	}
	
	GetRemove() { return this.m_remove;	 }
	GetKeyChange() {	return this.m_keyChange; }
	ClearKeyChange() { this.m_keyChange = "";	 }
	SetShow(_show) { this.m_show = _show; }
	GetShow() { return this.m_show;	 }
	GetCRenObjType()
	{
		return DfCRenObj.Null;
	}
	GetOffset()
	{
		return this.m_offset;
	}
	GetCPaint()
	{
		var dummy=new Array();
		return dummy;
	}
	GetKey() { return this.m_key; }
	SetKey(_key) 
	{
		if(this.m_key==_key)
			return;
		if(this.m_key.isEmpty()==false)
			this.m_keyChange = this.m_key;
		this.m_key = _key; 	
	}
	CanvasKeySet(_key) { this.m_key = _key;	 }
	Serialize()
	{
		var pac=new CPacket();
		pac.Push(this.m_offset);
		pac.Push(this.m_key);
		return pac;
	}
	Deserialize(_str)
	{
		var pac=new CPacket();
		pac.Deserialize(_str);
		this.m_offset = pac.GetInt32(0);
		this.m_key = pac.GetString(1);
	}
};

